src/effects/backends/builtin/echoeffect.cpp

```diff
@@ -118,6 +118,17 @@ void EchoEffect::loadEngineEffectParameters(
     m_pTripletParameter = parameters.value("triplet");
 }
 
+float averageSampleEnergy(SINT delayBufferSize, mixxx::SampleBuffer const& delay_buffer) {
+    float differenceSum = 0.0f;
+    for (SINT i = 0; i < delayBufferSize;
+            i = i + 16) { // We don't need to consider *all* samples,
```

Comment:
Are you sure? This just seems like downsampling that would introduce artifacts and miss high frequency signals. Is this just an optimization?

Reply:
good point, let's start with the simple thing considering all sample. Was a premature optimization-attempt

Reply:
reverted to `i = i + 1`